## Importing the Libraries which is needed for performing the analysis

In [1]:
import matplotlib.pyplot as plt
import numpy as np
from sklearn.linear_model import LogisticRegression ## for performing the logistic Regression
from sklearn.metrics import accuracy_score ## to obtain the accuracy score
import pandas as pd


In [2]:
redwine=pd.read_csv("wine-quality-red.csv", sep=",")
redwine

fixed_acidity  volatile_acidity  citric_acid  residual_sugar  chlorides  \
0               7.4             0.700         0.00             1.9      0.076   
1               7.8             0.880         0.00             2.6      0.098   
2               7.8             0.760         0.04             2.3      0.092   
3              11.2             0.280         0.56             1.9      0.075   
4               7.4             0.700         0.00             1.9      0.076   
5               7.4             0.660         0.00             1.8      0.075   
6               7.9             0.600         0.06             1.6      0.069   
7               7.3             0.650         0.00             1.2      0.065   
8               7.8             0.580         0.02             2.0      0.073   
9               7.5             0.500         0.36             6.1      0.071   
10              6.7             0.580         0.08             1.8      0.097   
11              7.5             0.500         0.36             6.1      0.071   
12              5.6             0.615         0.00             1.6      0.089   
13              7.8             0.610         0.29             1.6      0.114   
14              8.9             0.620         0.18             3.8      0.176   
15              8.9             0.620         0.19             3.9      0.170   
16              8.5             0.280         0.56             1.8      0.092   
17              8.1             0.560         0.28             1.7      0.368   
18              7.4             0.590         0.08             4.4      0.086   
19              7.9             0.320         0.51             1.8      0.341   
20              8.9             0.220         0.48             1.8      0.077   
21              7.6             0.390         0.31             2.3      0.082   
22              7.9             0.430         0.21             1.6      0.106   
23              8.5             0.490         0.11             2.3      0.084   
24              6.9             0.400         0.14             2.4      0.085   
25              6.3             0.390         0.16             1.4      0.080   
26              7.6             0.410         0.24             1.8      0.080   
27              7.9             0.430         0.21             1.6      0.106   
28              7.1             0.710         0.00             1.9      0.080   
29              7.8             0.645         0.00             2.0      0.082   
...             ...               ...          ...             ...        ...   
1569            6.2             0.510         0.14             1.9      0.056   
1570            6.4             0.360         0.53             2.2      0.230   
1571            6.4             0.380         0.14             2.2      0.038   
1572            7.3             0.690         0.32             2.2      0.069   
1573            6.0             0.580         0.20             2.4      0.075   
1574            5.6             0.310         0.78            13.9      0.074   
1575            7.5             0.520         0.40             2.2      0.060   
1576            8.0             0.300         0.63             1.6      0.081   
1577            6.2             0.700         0.15             5.1      0.076   
1578            6.8             0.670         0.15             1.8      0.118   
1579            6.2             0.560         0.09             1.7      0.053   
1580            7.4             0.350         0.33             2.4      0.068   
1581            6.2             0.560         0.09             1.7      0.053   
1582            6.1             0.715         0.10             2.6      0.053   
1583            6.2             0.460         0.29             2.1      0.074   
1584            6.7             0.320         0.44             2.4      0.061   
1585            7.2             0.390         0.44             2.6      0.066   
1586            7.5             0.310

## Defining the variables:

1.Fixed Acids in wine- Tartaric, Malic, citirc and Succinic acids.

2.Volatile acidity - it is a steam distillable acids present in wine. Primarily it has acetic acid but also lactic, formic, butyric and Propionic acids where hese compounds have much lower sensory threshold than acetic acid—both acetaldehyde and ethyl acetate are detectable at less than #200 mg/L in wine.

~U.S. legal limits of Volatile Acidity: 
Red Table Wine #1.2 g/L

3.Citric acid- it has a concentration nearly to #1/20 that of tartaric acid.

4.Residual Sugar from natural grape sugars leftover in a wine after the alcoholic fermentation finishes.  So for a wine with 10 grams per liter of residual sugar has 1% sweetness or a total of ~1.8 carbohydrates per serving (5 ounces / 150 g).

5.chlorides- The amount of salt in the wine.Total sulfur dioxide: amount of free and bound forms of S02; in low concentrations, SO2 is mostly undetectable in wine, but at free SO2 concentrations over #50 ppm, SO2 becomes evident in the nose and taste of wine. ppm = mass solute (mg) ÷ volume solution (L) so it is measured in #mgs

6,7.free and Total sulpur dioxide- It divides into parts with the major ones being the molecular form and the
bisulfite form. Certain “free” SO2 levels are needed to insure the target amounts of molecular SO2 are maintained. Rule of thumb to use Sulpur dioxide is 1.5 grams (1/4 tsp.) per five (5) gallons for 2 or 3 times the wine is racked.

total SO2 of #0.015 g/litre in red wines. 0.005 g/l  of molecular SO2 is considered the needed amount for red wine.

#### It is mandatory to include 'contains sulphites' on the label if total SO2 is over 0.0001 g/litre.

8.Density- i am asuming as i have taken #mg in chlorides i use the matrix of #mg and cubic millimeter where density = 1.01 in actual and we have the max =1.0036

#### where cubic millimeter= 1 littre
9.pH - has scale of 0 -14 and measure of hydrogen ion concentration.

10.Sulphate- The maximum levels of sulfur dioxide that a wine can contain #160 ppm for red wine

11.Alchol- The alcohol content usually ranges from #12–15%

12.Class- it is the score ranges form #0-10

## Source:
https://waterhouse.ucdavis.edu/whats-in-wine/volatile-acidity#:~:targetText=Volatile%20acidity%20refers%20to%20the,%2C%20butyric%2C%20and%20propionic%20acids.&targetText=These%20compounds%20have%20much%20lower,200%20mg%2FL%20in%20wine. 

https://www.aqua-calc.com/page/density-table/substance/vinegar-coma-and-blank-red-blank-wine

https://www.healthline.com/nutrition/red-wine-good-or-bad

http://www.piwine.com/media/home-wine-making-basics/using_sulfur_dioxide.pdf

# step1: Performance of intial Exploratory Data Aanalysis

In [3]:
redwine.columns

Index(['fixed_acidity', 'volatile_acidity', 'citric_acid', 'residual_sugar',
       'chlorides', 'free_sulfur_dioxide', 'total_sulfur_dioxide', 'density',
       'pH', 'sulphates', 'alcohol', 'class'],
      dtype='object')

In [4]:
redwine.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
fixed_acidity           1599 non-null float64
volatile_acidity        1599 non-null float64
citric_acid             1599 non-null float64
residual_sugar          1599 non-null float64
chlorides               1599 non-null float64
free_sulfur_dioxide     1599 non-null float64
total_sulfur_dioxide    1599 non-null float64
density                 1599 non-null float64
pH                      1599 non-null float64
sulphates               1599 non-null float64
alcohol                 1599 non-null float64
class                   1599 non-null int64
dtypes: float64(11), int64(1)
memory usage: 150.0 KB


### In the data there is no catagroical variables to study but we have all numerical variables

In [5]:
redwine.describe()

fixed_acidity  volatile_acidity  citric_acid  residual_sugar  \
count    1599.000000       1599.000000  1599.000000     1599.000000   
mean        8.319637          0.527821     0.270976        2.538806   
std         1.741096          0.179060     0.194801        1.409928   
min         4.600000          0.120000     0.000000        0.900000   
25%         7.100000          0.390000     0.090000        1.900000   
50%         7.900000          0.520000     0.260000        2.200000   
75%         9.200000          0.640000     0.420000        2.600000   
max        15.900000          1.580000     1.000000       15.500000   

         chlorides  free_sulfur_dioxide  total_sulfur_dioxide      density  \
count  1599.000000          1599.000000           1599.000000  1599.000000   
mean      0.087467            15.874922             46.467792     0.996747   
std       0.047065            10.460157             32.895324     0.001887   
min       0.012000             1.000000              6.000000     0.990070   
25%       0.070000             7.000000             22.000000     0.995600   
50%       0.079000            14.000000             38.000000     0.996750   
75%       0.090000            21.000000             62.000000     0.997835   
max       0.611000            72.000000            289.000000     1.003690   

                pH    sulphates      alcohol        class  
count  1599.000000  1599.000000  1599.000000  1599.000000  
mean      3.311113     0.658149    10.422983     5.636023  
std       0.154386     0.169507     1.065668     0.807569  
min       2.740000     0.330000     8.400000     3.000000  
25%       3.210000     0.550000     9.500000     5.000000  
50%       3.310000     0.620000    10.200000     6.000000  
75%       3.400000     0.730000    11.100000     6.000000  
max       4.010000     2.000000    14.900000     8.000000

#### so we can see the deviation "0. " differentiation. Except for Free_sulfur_dioxide and Total_sulpur_dioxide

## Variance of Numerical Columns

In [6]:
redwine.var()

fixed_acidity              3.031416
volatile_acidity           0.032062
citric_acid                0.037947
residual_sugar             1.987897
chlorides                  0.002215
free_sulfur_dioxide      109.414884
total_sulfur_dioxide    1082.102373
density                    0.000004
pH                         0.023835
sulphates                  0.028733
alcohol                    1.135647
class                      0.652168
dtype: float64

### Performing normalisation

In [7]:
def normalize(wine):
    result = wine.copy()
    for feature_name in wine.columns:
        max_value = wine[feature_name].max()
        min_value = wine[feature_name].min()
        result[feature_name] = (wine[feature_name] - min_value) / (max_value - min_value)
    return result

In [8]:
red=normalize(redwine)

In [9]:
red.head()

fixed_acidity  volatile_acidity  citric_acid  residual_sugar  chlorides  \
0       0.247788          0.397260         0.00        0.068493   0.106845   
1       0.283186          0.520548         0.00        0.116438   0.143573   
2       0.283186          0.438356         0.04        0.095890   0.133556   
3       0.584071          0.109589         0.56        0.068493   0.105175   
4       0.247788          0.397260         0.00        0.068493   0.106845   

   free_sulfur_dioxide  total_sulfur_dioxide   density        pH  sulphates  \
0             0.140845              0.098940  0.567548  0.606299   0.137725   
1             0.338028              0.215548  0.494126  0.362205   0.209581   
2             0.197183              0.169611  0.508811  0.409449   0.191617   
3             0.225352              0.190813  0.582232  0.330709   0.149701   
4             0.140845              0.098940  0.567548  0.606299   0.137725   

    alcohol  class  
0  0.153846    0.4  
1  0.215385    0.4  
2  0.215385    0.4  
3  0.215385    0.6  
4  0.153846    0.4

# Step2: Performance Classification

In [10]:
x = redwine.iloc[:,[0,1,2,3,4,5,6,7,8,9,10]].values
y = redwine.iloc[:,11].values

In [11]:
x

array([[ 7.4  ,  0.7  ,  0.   , ...,  3.51 ,  0.56 ,  9.4  ],
       [ 7.8  ,  0.88 ,  0.   , ...,  3.2  ,  0.68 ,  9.8  ],
       [ 7.8  ,  0.76 ,  0.04 , ...,  3.26 ,  0.65 ,  9.8  ],
       ...,
       [ 6.3  ,  0.51 ,  0.13 , ...,  3.42 ,  0.75 , 11.   ],
       [ 5.9  ,  0.645,  0.12 , ...,  3.57 ,  0.71 , 10.2  ],
       [ 6.   ,  0.31 ,  0.47 , ...,  3.39 ,  0.66 , 11.   ]])

In [12]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.25,random_state=0)

# Classifing the variables using Logistic Regression

In [13]:
classifier=LogisticRegression(random_state=0)
classifier.fit(x_train,y_train)

C:\Users\Nivya Sree\Anaconda3\anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
C:\Users\Nivya Sree\Anaconda3\anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.
  "this warning.", FutureWarning)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='warn',
          n_jobs=None, penalty='l2', random_state=0, solver='warn',
          tol=0.0001, verbose=0, warm_start=False)

In [14]:
y_pred=classifier.predict(x_test)

In [15]:
y_pred

array([6, 5, 6, 5, 6, 5, 5, 6, 5, 5, 5, 5, 6, 5, 6, 6, 7, 6, 6, 5, 6, 5,
       6, 5, 5, 5, 5, 6, 5, 6, 6, 6, 6, 5, 6, 6, 5, 5, 6, 6, 5, 6, 7, 6,
       6, 5, 6, 6, 5, 6, 5, 5, 6, 6, 5, 5, 5, 5, 6, 5, 5, 6, 6, 6, 5, 6,
       5, 6, 6, 6, 5, 5, 5, 5, 5, 6, 5, 5, 5, 6, 5, 5, 6, 6, 6, 5, 6, 5,
       5, 5, 5, 5, 6, 5, 6, 5, 6, 5, 5, 6, 7, 6, 6, 6, 5, 5, 5, 5, 6, 5,
       6, 5, 6, 5, 6, 6, 6, 6, 6, 6, 5, 6, 6, 5, 6, 6, 5, 5, 6, 6, 5, 5,
       6, 6, 6, 5, 6, 5, 6, 5, 6, 5, 6, 5, 6, 6, 6, 6, 6, 5, 6, 6, 5, 6,
       6, 6, 5, 5, 6, 6, 6, 6, 6, 5, 6, 5, 6, 6, 5, 6, 6, 5, 5, 7, 6, 6,
       7, 7, 6, 5, 5, 7, 5, 6, 7, 5, 6, 6, 5, 6, 6, 6, 5, 5, 5, 5, 5, 5,
       5, 5, 5, 6, 5, 5, 5, 5, 5, 6, 6, 5, 6, 5, 6, 7, 5, 5, 6, 6, 6, 5,
       5, 6, 6, 6, 5, 6, 6, 6, 5, 5, 5, 6, 5, 6, 6, 6, 6, 6, 6, 6, 5, 5,
       5, 5, 6, 5, 5, 5, 5, 6, 5, 5, 5, 5, 6, 5, 5, 5, 5, 5, 5, 6, 5, 6,
       5, 5, 5, 5, 6, 6, 5, 6, 6, 6, 6, 6, 6, 5, 7, 6, 5, 6, 6, 6, 6, 5,
       5, 5, 6, 6, 6, 6, 6, 5, 5, 6, 5, 5, 5, 5, 6,

In [16]:
accuracy = accuracy_score(y_test,y_pred)

print(accuracy)

0.62


## Using of Neural Networks 

In [17]:
from sklearn.neural_network import MLPClassifier

model = MLPClassifier(hidden_layer_sizes=(200,100), max_iter=1000)
model.fit(x, y)

MLPClassifier(activation='relu', alpha=0.0001, batch_size='auto', beta_1=0.9,
       beta_2=0.999, early_stopping=False, epsilon=1e-08,
       hidden_layer_sizes=(200, 100), learning_rate='constant',
       learning_rate_init=0.001, max_iter=1000, momentum=0.9,
       n_iter_no_change=10, nesterovs_momentum=True, power_t=0.5,
       random_state=None, shuffle=True, solver='adam', tol=0.0001,
       validation_fraction=0.1, verbose=False, warm_start=False)

In [18]:
predicted_class=model.predict(x_test)

## Accuracy of the model

In [19]:
accuracy1 = accuracy_score(y_test,predicted_class)

print(accuracy1)

0.665


# Confusion Matrix

In [20]:
from sklearn.metrics import confusion_matrix

cm=confusion_matrix(y_test,y_pred)
print(cm)

[[  0   0   2   0   0   0]
 [  0   0   9   5   0   0]
 [  0   0 131  37   1   0]
 [  0   0  52 113   5   0]
 [  0   0   3  33   4   0]
 [  0   0   0   3   2   0]]


# Kinds of Nureal Network

In [21]:
nnmodel = MLPClassifier(hidden_layer_sizes=(30,50), max_iter=1000)

In [22]:
nnmodel.fit(x_train,y_train)

MLPClassifier(activation='relu', alpha=0.0001, batch_size='auto', beta_1=0.9,
       beta_2=0.999, early_stopping=False, epsilon=1e-08,
       hidden_layer_sizes=(30, 50), learning_rate='constant',
       learning_rate_init=0.001, max_iter=1000, momentum=0.9,
       n_iter_no_change=10, nesterovs_momentum=True, power_t=0.5,
       random_state=None, shuffle=True, solver='adam', tol=0.0001,
       validation_fraction=0.1, verbose=False, warm_start=False)

In [23]:
nnpredicted_class=nnmodel.predict(x_test)
accuracy_score(y_test,nnpredicted_class)

0.615

In [24]:
from sklearn.metrics import classification_report

report=classification_report(y_test,nnpredicted_class)
print(report)

              precision    recall  f1-score   support

           3       0.00      0.00      0.00         2
           4       0.00      0.00      0.00        14
           5       0.65      0.74      0.69       169
           6       0.60      0.60      0.60       170
           7       0.53      0.47      0.50        40
           8       0.00      0.00      0.00         5

   micro avg       0.61      0.61      0.61       400
   macro avg       0.30      0.30      0.30       400
weighted avg       0.58      0.61      0.60       400



C:\Users\Nivya Sree\Anaconda3\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


In [25]:
nn_cm=confusion_matrix(y_test, nnpredicted_class)
print(nn_cm)

[[  0   0   2   0   0   0]
 [  0   0   9   3   2   0]
 [  0   0 125  44   0   0]
 [  0   0  55 102  13   0]
 [  0   0   2  19  19   0]
 [  0   0   0   3   2   0]]


In [26]:
nnmodel1 = MLPClassifier(hidden_layer_sizes=(40,50), max_iter=1000)

In [27]:
nnmodel1.fit(x_train,y_train)

nnpredicted_class1=nnmodel1.predict(x_test)
accuracy_score(y_test,nnpredicted_class)

0.615

In [28]:
nn_cm1=confusion_matrix(y_test, nnpredicted_class1)
print(nn_cm1)

[[  0   0   2   0   0   0]
 [  0   0   9   5   0   0]
 [  0   0 125  43   1   0]
 [  0   0  59  94  17   0]
 [  0   0   3  18  19   0]
 [  0   0   1   2   2   0]]


In [29]:
nnmodel2 = MLPClassifier(hidden_layer_sizes=(100,200),max_iter=1000)

In [30]:
nnmodel2.fit(x_train,y_train)

nnpredicted_class2=nnmodel2.predict(x_test)
accuracy_score(y_test,nnpredicted_class2)

0.615

In [31]:
nnmodel3 = MLPClassifier(hidden_layer_sizes=(200,200), max_iter=1000)

In [32]:
nnmodel3.fit(x_train,y_train)

nnpredicted_class3=nnmodel3.predict(x_test)
accuracy_score(y_test,nnpredicted_class3)

0.555

In [33]:
nnmodel4 = MLPClassifier(hidden_layer_sizes=(400,400),max_iter=1000)

In [34]:
nnmodel4.fit(x_train,y_train)

nnpredicted_class4=nnmodel4.predict(x_test)
accuracy_score(y_test,nnpredicted_class4)

0.6025

In [35]:
nnmodel5 = MLPClassifier(hidden_layer_sizes=(800,1000))

nnmodel5.fit(x_train,y_train)

nnpredicted_class5=nnmodel4.predict(x_test)
accuracy_score(y_test,nnpredicted_class5)

0.6025

## Accuracy model :

In Redwine data when I tried to predict the model and try to understand the noramlisation of the class variables then we found Accuracy is 0.62

When we performed Nureal networks with the hidden_layer_sizes=(200,100), we found Accuracy is 0.6425 and max_iter=1000 and kept it constant.

When I started increasing the size of hidden_layer_sizes=(30,50), I found Accuracy is 0.5725 

When i changed the hidden_layer_sizes=(40,50), I found interstingly the Accuracy is 0.5725

When i changed the hidden_layer_sizes=(100,200), I found Accuracy is 0.6125

when i try to increase the hidden_layer_sizes=(200,200)/ (400,400)/ (800,1000), I found Accuracy is constant or ranging 0.57-0.56

## Implementing MINMAX 

In [36]:
from sklearn.preprocessing import MinMaxScaler

In [37]:
MinMaxScaler(feature_range=(0, 1), copy=True)

MinMaxScaler(copy=True, feature_range=(0, 1))

In [38]:
minmax=MinMaxScaler(feature_range=(0, 1), copy=True)

## Performance of minmax for whole dataset

In [42]:
scale=minmax.fit_transform(redwine)

C:\Users\Nivya Sree\Anaconda3\anaconda3\lib\site-packages\sklearn\preprocessing\data.py:334: DataConversionWarning: Data with input dtype int64, float64 were all converted to float64 by MinMaxScaler.
  return self.partial_fit(X, y)


In [43]:
scale

array([[0.24778761, 0.39726027, 0.        , ..., 0.13772455, 0.15384615,
        0.4       ],
       [0.28318584, 0.52054795, 0.        , ..., 0.20958084, 0.21538462,
        0.4       ],
       [0.28318584, 0.43835616, 0.04      , ..., 0.19161677, 0.21538462,
        0.4       ],
       ...,
       [0.15044248, 0.26712329, 0.13      , ..., 0.25149701, 0.4       ,
        0.6       ],
       [0.11504425, 0.35958904, 0.12      , ..., 0.22754491, 0.27692308,
        0.4       ],
       [0.12389381, 0.13013699, 0.47      , ..., 0.19760479, 0.4       ,
        0.6       ]])

In [44]:
x = redwine.iloc[:,[0,1,2,3,4,5,6,7,8,9,10]].values
y = redwine.iloc[:,11].values

In [47]:
x

array([[ 7.4  ,  0.7  ,  0.   , ...,  3.51 ,  0.56 ,  9.4  ],
       [ 7.8  ,  0.88 ,  0.   , ...,  3.2  ,  0.68 ,  9.8  ],
       [ 7.8  ,  0.76 ,  0.04 , ...,  3.26 ,  0.65 ,  9.8  ],
       ...,
       [ 6.3  ,  0.51 ,  0.13 , ...,  3.42 ,  0.75 , 11.   ],
       [ 5.9  ,  0.645,  0.12 , ...,  3.57 ,  0.71 , 10.2  ],
       [ 6.   ,  0.31 ,  0.47 , ...,  3.39 ,  0.66 , 11.   ]])

In [48]:
y=np.reshape(y,newshape= (-1,1))
y

array([[5],
       [5],
       [5],
       ...,
       [6],
       [5],
       [6]], dtype=int64)

In [49]:
x=minmax.fit_transform(x)
x

array([[0.24778761, 0.39726027, 0.        , ..., 0.60629921, 0.13772455,
        0.15384615],
       [0.28318584, 0.52054795, 0.        , ..., 0.36220472, 0.20958084,
        0.21538462],
       [0.28318584, 0.43835616, 0.04      , ..., 0.40944882, 0.19161677,
        0.21538462],
       ...,
       [0.15044248, 0.26712329, 0.13      , ..., 0.53543307, 0.25149701,
        0.4       ],
       [0.11504425, 0.35958904, 0.12      , ..., 0.65354331, 0.22754491,
        0.27692308],
       [0.12389381, 0.13013699, 0.47      , ..., 0.51181102, 0.19760479,
        0.4       ]])

In [50]:
np.asarray(x, dtype=float)

array([[0.24778761, 0.39726027, 0.        , ..., 0.60629921, 0.13772455,
        0.15384615],
       [0.28318584, 0.52054795, 0.        , ..., 0.36220472, 0.20958084,
        0.21538462],
       [0.28318584, 0.43835616, 0.04      , ..., 0.40944882, 0.19161677,
        0.21538462],
       ...,
       [0.15044248, 0.26712329, 0.13      , ..., 0.53543307, 0.25149701,
        0.4       ],
       [0.11504425, 0.35958904, 0.12      , ..., 0.65354331, 0.22754491,
        0.27692308],
       [0.12389381, 0.13013699, 0.47      , ..., 0.51181102, 0.19760479,
        0.4       ]])

In [51]:
type(x)

numpy.ndarray

In [52]:
X_train,X_test,Y_train,Y_test=train_test_split(x,y,test_size=0.25,random_state=0)

In [ ]:
scalar = MLPClassifier(hidden_layer_sizes=(200,100), max_iter=1000)
print(scalar.fit(x,y))
scalar_pred_class=scalar.predict(X_test)
print(scalar_pred_class)

C:\Users\Nivya Sree\Anaconda3\anaconda3\lib\site-packages\sklearn\neural_network\multilayer_perceptron.py:916: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [54]:
ac_scale = accuracy_score(Y_test,scalar_pred_class)

print(ac_scale)

0.7175


## Accuarcy after usage of MINMAX Scale

By using the minmax the accuracy of the data is has increased from 0.6425 to 0.7175 for the same neural network classifier having hidden_layer_sizes=(200,100), max_iter=1000